pythia/cleaning/midnight_rotation.py

```diff
@@ -0,0 +1,111 @@
+import datetime
+
+import astropy.units as u
+from sunpy.physics.differential_rotation import diff_rot
+
+__all__ = ['MidnightRotation']
+
+
+class MidnightRotation:
+    """
+    A class to rotate the coordinates of an observation
+    to its coordinates at the nearest midnight.
+    """
+    def get_nearest_midnight(self, obsdate: str, fmt='%Y-%m-%d %H:%M:%S'):
+        """
+        For a given observation time and date, returns the datetime of the nearest midnight.
+
+        Parameters
+        ----------
+        obsdate : str
+            The observation time and date.
+        fmt : str, optional
+            The format in which obsdate is represented, by default '%Y-%m-%d %H:%M:%S'
+
+        Returns
+        -------
+        nearest_midnight : datetime.datetime
+            The datetime of the nearest midnight.
+
+        Examples
+        --------
+        >>> from pythia.cleaning.midnight_rotation import MidnightRotation
+        >>> midnight_rotation = MidnightRotation()
+        >>> obsdate = '2000-01-01 12:47:02'
+        >>> midnight_rotation.get_nearest_midnight(obsdate)
+        datetime.datetime(2000, 1, 2, 0, 0)
+        >>> obsdate = '2000-01-01 11:47:02'
+        >>> midnight_rotation.get_nearest_midnight(obsdate)
+        datetime.datetime(2000, 1, 1, 0, 0)
+        """
+        current = datetime.datetime.strptime(obsdate, fmt)
+        if current.hour >= 12:
+            current = current + datetime.timedelta(days=1)
+        return current.replace(hour=0, minute=0, second=0, microsecond=0)
+
+    def get_seconds_to_nearest_midnight(self, obsdate: str, fmt='%Y-%m-%d %H:%M:%S'):
+        """
+        For a given observation time and date, returns the seconds to the nearest midnight.
+
+        Parameters
+        ----------
+        obsdate : str
+            The observation time and date.
+        fmt : str, optional
+            The format in which obsdate is represented, by default '%Y-%m-%d %H:%M:%S'
+
+        Returns
+        -------
+        seconds_to_nearest_midnight : int
+            Returns the seconds to the nearest midnight.
+            Returned seconds are negative if the nearest midnight is of the same day.
+
+        Examples
+        --------
+        >>> from pythia.cleaning.midnight_rotation import MidnightRotation
+        >>> midnight_rotation = MidnightRotation()
+        >>> obsdate = '2000-01-01 12:47:02'
+        >>> midnight_rotation.get_seconds_to_nearest_midnight(obsdate)
+        40378
+        >>> obsdate = '2000-01-01 11:47:02'
+        >>> midnight_rotation.get_seconds_to_nearest_midnight(obsdate)
+        -42422
+        """
+        timedifference = datetime.datetime.strptime(obsdate, fmt) - self.get_nearest_midnight(obsdate, fmt)
+        if timedifference.days < 0:
+            return 86400 - timedifference.seconds
+        return -1 * timedifference.seconds
```

Comment:
```suggestion
        return timedifference.total_seconds()
```
Wouldn't this be the same?